ice40/utils/fasm_icebox/fasm_icebox_utils.py

```diff
@@ -0,0 +1,469 @@
+"""Convert the icestorm database to a prjxray type DB
+
+The goal is that the icestorm db doesn't encode information about BRAM init or
+inversion of some bits such as RamConfig PowerUp (differs between 8k and 1k?)
+
+This is handled pretty wonkley in explain and hlc code
+
+These are not handled as features in icebox
+ - break out INIT for ram
+ - break out bits for LC_[0-7] INIT
+
+
+Another name variant
+
+HLC introduce a globally unique name for each wire named on the right endpoint.
+FASM will have a slight variation for some features due to not supporting / character
+
+"""
+import numpy as np
```

Comment:
numpy doesn't seem needed here?  Can you explain why it is being used over plain python types?

Reply:
I think it simplifies the code using numpy arrays. I'd actually like to push this down to icestorm. icebox stores the bits as a list of strings which this goes back and forth between. If you feel strongly I switch to 2D lists.